### ISSUE TITLE
Erreur sur certains dossiers

### ISSUE THREAD

**Issue by SDIS91** [SDIS62/prevarisc#586]
Bonjour

Après la reprise des données de Winprev, quand on cliques sur certains dossiers, une erreur SQL s'affiche. Tous les dossiers ne sont pas concernés par cette erreur donc le probleme vient probablement de nos donnees. Au vue de l'erreur, savez-vous ou je dois regarder?

![dossier](https://cloud.githubusercontent.com/assets/15104477/15533499/2a87ee72-2265-11e6-8e88-facb0ffc48b8.png)

Merci d'avance

**Comment by aechinard** [comment]
Bonjour,

Même problème que vous. Cela fonctionne correctement quand on enlève le dossier qui pose problème dans la table etablissementdossier mais avec ce dossier en moins. Je ne comprends pas d’où cela vient car toutes les infos sont correctes et cela fonctionne sur les autres dossiers...
Si vous avez des infos, je suis preneur.

**Comment by aechinard** [comment]
Bonjour,

J'ai résolu le problème chez moi. Il faut modifier une valeur dans la table Dossier, Createur_dossier pour le dossier qui pose souci. Chez vous ce doit-être le 6
UPDATE `prevarisc`.`dossier` SET `CREATEUR_DOSSIER` = NULL WHERE `dossier`.`ID_DOSSIER` = 6;

Dans mon cas les valeurs étaient à zéro, je les ai passées à NULL avec phpmyadmin.

**Comment by SDIS91** [comment]
Bonjour

Merci pour cette réponse, ça marche pour certains dossiers mais pas pour d'autres (pour le 6 par ex ça n'a pas marché). D'alleurs, il y a des dossiers qui posent problème même quand CREATEUR_DOSSIER est different de zéro.

Cordialement

**Comment by aechinard** [comment]
Bonjour,

Si vous avez beaucoup de champs avec des valeurs à zéro (00-00-0000 00:00:00, 0...) essayez de les mettre à NULL, il est possible que cela vienne de là...
Vous avez migré vos données par quel moyen ?

Cordialement

**Comment by SDIS91** [comment]
Bonjour

Toutes les dates dans la table sont soit bien renseignées soit NULL.
Nous avons migré en nous aidant de la procedure "reprise de données" (https://github.com/SDIS62/prevarisc/issues/357#issuecomment-82180724) mais un peu retravaillée par nous.

**Comment by aechinard** [comment]
Bonjour,

Je viens d'avoir le soucis sur un autre établissement. Celui-ci comporte 3 dossiers (2839, 4633 et 8988). J'ai mis la valeur CREATEUR_DOSSIER à NULL pour le 2839 => problème toujours présent, j'ai donc mis la même valeur à NULL sur le dossier suivant 4633 et là tout refonctionne... Avez-vous fait de même avec tous les dossiers de l'établissement concerné ?
Ce problème est vraiment très bizarre...

**Comment by SDIS91** [comment]
Bonjour

Nous n'avons toujours pas trouvé la cause de l'erreur. Tous les champs CREATEUR_DOSSIER qui contenaient zéro on été mis à NULL dans la table dossiers. Quelques dossiers ont été corrigés mais pas d'autres, donc la cause bien que liée à l'ID de l'utilisateur est ailleurs (probablement dans la requête:
`$row = $DB_utilisateursInfo->fetchRow("ID_UTILISATEURINFORMATIONS = " . $DB_utilisateurs->find($fiche["UTILISATEUR_ETABLISSEMENTINFORMATIONS"])->current()->ID_UTILISATEURINFORMATIONS)->toArray();`
dans le fichier /application/services/Etablissement.php

**Comment by kdubuc** [comment]
Effectivement, si un dossier pose un soucis au niveau requête, les autres dossiers liés de près ou de loin à ce dernier seront aussi affectés.

CREATEUR_DOSSIER  et UTILISATEUR_ETABLISSEMENTINFORMATIONS sont des clés étrangères vers la table UTILISATEURINFORMATIONS, utilisée pour gérer des coordonnées. Mettre à NULL sur  CREATEUR_DOSSIER  à tous les dossiers devrait régler le problème.

**Comment by SDIS91** [comment]
Bonjour

Merci pour la réponse. J'avais mis CREATEUR_DOSSIER à NULL il y a quelques jours et ça avait corrigé quelques dossiers. Or aujourd'hui, un échantillon des dossiers à problème me montre que tout est réglé. J'avais fait un git pull au préalable, est-ce que ce sont les commits de cette nuit qui ont réglé le problème ou est-ce une histoire de cache qui a coincidé avec le pull?

Encore merci

**Comment by aechinard** [comment]
Bonjour,

Il est bien possible que ce soit le cas... Cela semblait être un problème de requête.

Tant mieux si tout est ok chez vous.

Au plaisir

**Comment by SDIS91** [comment]
Bonjour

En fait nous avon encore quelques dossiers qui affichent l'erreur bien que tous les champs CREATEUR_DOSSIER qui étaient 0 soient NULL.
Et je ne trouve rien de remarquable concernant ces dossiers pour pouvoir cerner la cause.

Cordialement

**Comment by regnaulto** [comment]
bonjour,

pour information et après quelques tests, cette erreur apparait également sur action sur le bouton "Afficher l'historique"

Il semblerait qu'il y ai un rapport avec la function getHistorique présente dans application\services\Etablissement.php

Les recherches continuent ....

**Comment by SDIS91** [comment]
Bonjour

Pour information, nous avons finalement réglé le problème. En effet, certains dossiers avaient un CREATEUR_DOSSIER dont l'identifiant n'existait pas dans la table utilisateur.
Si ça peut servir à quelqu'un.

Cordialement
